test/lib/absinthe/extensions_test.exs

```diff
@@ -0,0 +1,47 @@
+defmodule Absinthe.ExtensionsTest do
+  use Absinthe.Case, async: true
+
+  defmodule Schema do
+    use Absinthe.Schema
+
+    query do
+      field :foo, :string do
+        middleware :resolve_foo
+      end
+    end
+
+    def resolve_foo(res, _opts) do
+      %{res |
+        value: "hello world",
+        state: :resolved,
+        extensions: %{foo: 1},
```

Comment:
arbitrary values can be placed here by middleware. All middleware share a single map, so it may be wise to namespace it by the name of the middleware.